http-client/src/main/java/com/facebook/airlift/http/client/thrift/ThriftResponse.java

```diff
@@ -0,0 +1,71 @@
+package com.facebook.airlift.http.client.thrift;
+
+import com.facebook.airlift.http.client.HeaderName;
+import com.google.common.collect.ImmutableListMultimap;
+import com.google.common.collect.ListMultimap;
+
+import javax.annotation.Nullable;
+
+import java.util.List;
+
+import static java.util.Objects.requireNonNull;
+
+public class ThriftResponse<T>
+{
+    private final int statusCode;
+    private final String statusMessage;
+    private final ListMultimap<HeaderName, String> headers;
+    private final T value;
+    private final IllegalArgumentException exception;
+
+    ThriftResponse(
+            int statusCode,
+            String statusMessage,
+            ListMultimap<HeaderName, String> headers,
+            T value,
+            IllegalArgumentException exception)
+    {
+        this.statusCode = statusCode;
+        this.statusMessage = requireNonNull(statusMessage, "statusMessage is null");
+        this.headers = headers != null ? ImmutableListMultimap.copyOf(headers) : null;
+        this.value = value;
+        this.exception = exception;
+    }
+
+    public int getStatusCode()
+    {
+        return statusCode;
+    }
+
+    public String getStatusMessage()
+    {
+        return statusMessage;
+    }
+
+    public T getValue()
+    {
+        return value;
+    }
+
+    @Nullable
+    public String getHeader(String name)
+    {
+        List<String> values = getHeaders().get(HeaderName.of(name));
+        return values.isEmpty() ? null : values.get(0);
```

Comment:
Do we need to add a null check too?

Reply:
`values` cannot be null according to the javadocs.

```
Returns a view collection of the values associated with key in this multimap, if any. Note that when containsKey(key) is false, this returns an empty collection, not null.
```

Reply:
Cool!